.github/workflows/prototype-tests.yml

```
@@ -40,13 +40,18 @@ jobs:
         run: pip install --progress-bar=off pytest pytest-mock
 
       - name: Run prototype datasets tests
+        id: datasets
         shell: bash
         run: pytest --durations=20 test/test_prototype_datasets*.py
 
       - name: Run prototype transforms tests
+        id: transforms
+        if: ${{ success() || steps.datasets.conclusion == 'failure' }}
```

We cannot use

```suggestion
        if: always()
```

here for two reasons:

1. [`always()`](https://docs.github.com/en/actions/learn-github-actions/expressions#always) is true even if the workflow is cancelled. Meaning we can't the tests through the UI.
2. We only want to run the tests if the previous tests failed not for example the setup.

---

Hi @pmeier a noob question here, what is the reason we only want to run the test if the previous test failed?
Can't we do something like:
```
if: success() || failure()
```
(I get this from https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f )

---

`failure()` will be true if *any* step has failed. Imagine installing `torchvision` failed for some reason. If we use `failure()` the tests will now also be run and obviously fail. This is somewhat a waste of CI resources and also gives a false signal that there is also a problem with the tests (the step will be marked as failure rather than as skipped) although that is impossible to say.

---

I see, yeah I agree that it will give a wrong signal to the test which is not desired. Thanks for the explanation!